project1/server/agent.cpp

```diff
@@ -180,15 +182,18 @@ namespace server {
         }
 
         // monitor for termination request for get and put commands
-        uint16_t kBytesSent = 0;
-        const uint32_t kBytesToSend = 1000;
+        uint16_t total_bytes_sent = 0;
         bool terminated = false;
 
         // continue until we've sent the entire message
-        while (kBytesSent < outgoing_message_buffer_.size()) {
+        while (total_bytes_sent < outgoing_message_buffer_.size()) {
             if (!terminated) {
-                kBytesSent += send(client_fd_, outgoing_message_buffer_.data() + kBytesSent,
-                                   kBytesToSend, 0);
+                const uint32_t kBytesToSend = std::min( (int) outgoing_message_buffer_.size() - total_bytes_sent, 1000);
```

Comment:
This should be compared to `kBytesToSend`, instead of directly comparing to 1000. While we're at it, `kBytesToSend` is probably not the best name for this constant. I would suggest `kSendChunkSize` or something.